src/lib.rs

```diff
@@ -3361,23 +3357,30 @@ impl Build {
                     // here.
 
                     let compiler = self.get_base_compiler()?;
-                    let mut lib = String::new();
-                    if compiler.family == (ToolFamily::Msvc { clang_cl: true }) {
-                        // See if there is 'llvm-lib' next to 'clang-cl'
-                        // Another possibility could be to see if there is 'clang'
-                        // next to 'clang-cl' and use 'search_programs()' to locate
-                        // 'llvm-lib'. This is because 'clang-cl' doesn't support
-                        // the -print-search-dirs option.
-                        if let Some(mut cmd) = self.which(&compiler.path, None) {
-                            cmd.pop();
-                            cmd.push("llvm-lib.exe");
-                            if let Some(llvm_lib) = self.which(&cmd, None) {
-                                llvm_lib.to_str().unwrap().clone_into(&mut lib);
+                    let lib = if compiler.family == (ToolFamily::Msvc { clang_cl: true }) {
+                        self.search_programs(
+                            &compiler.path,
+                            Path::new("llvm-lib"),
+                            &self.cargo_output,
+                        )
+                        .or_else(|| {
+                            // See if there is 'llvm-lib' next to 'clang-cl'
+                            if let Some(mut cmd) = self.which(&compiler.path, None) {
+                                cmd.pop();
+                                cmd.push("llvm-lib.exe");
```

Comment:
The original remark was "here and elsewhere." This line is the "elsewhere."

Just in case, again, I personally have no problem with explicit .exe, but others might appreciate the consideration.

Reply:
Thanks updated, didn't realize it means this